core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -56,6 +56,10 @@ class BuildAvroProjection extends AvroCustomOrderSchemaVisitor<Schema, Schema.Fi
   @Override
   @SuppressWarnings("checkstyle:CyclomaticComplexity")
   public Schema record(Schema record, List<String> names, Iterable<Schema.Field> schemaIterable) {
+    if (current.isVariantType()) {
```

Comment:
I think that the corresponding `visit` method should be updated to call a new visitor `variant` method. That will be cleaner.

The visitor should look for the `variant` logical type, so we will need to implement the logical type.

Reply:
You are right. This is to workaround before the logical type is added in Avro by using Iceberg variant type. Let me add a comment for this.

Reply:
If we are updating the visitor implementations, I think we should also update the visit method. We can also delay making thes changes.

Reply:
You mean handle variant in the following visit method? As I mentioned that we don't have variant logical type in Schema.getType() from avro, we can't handle that for now. Let me know if understand correctly. I think this is needed for metadata change and  I can defer until then.

> abstract class AvroCustomOrderSchemaVisitor<T, F> {
>   public static <T, F> T visit(Schema schema, AvroCustomOrderSchemaVisitor<T, F> visitor) {
>     switch (schema.getType()) {
>